scylla-cdc/src/reader.rs

```diff
@@ -101,6 +102,78 @@ impl StreamReader {
 
         Ok(())
     }
+
+    pub async fn fetch_cdc_with_consumer(
+        &self,
+        keyspace: String,
+        table_name: String,
+        mut consumer: Box<dyn Consumer + Send>,
+    ) -> anyhow::Result<()> {
+        // [TODO]: This function is almost identical to fetch_cdc,
+        //      but it uses the Consumer trait as a callback.
+        //      The reason for leaving the old function are the tests,
+        //      there is no obvious solution as for what to do with them.
+
```

Comment:
This temporary solution is left, because I am not sure what to do with the old tests - if the old `fetch_cdc` function is removed, we will have to fix them.

Reply:
Won't just passing a lambda to `fetch_cdc` which calls the `consumer.consume_cdc` work? Something like that (in `CDCReaderWorker::run`):

```rust
let reader = Arc::clone(reader);
let keyspace = self.keyspace.clone();
let table_name = self.table_name.clone();
let mut consumer = self.consumer_factory.new_consumer();
tokio::spawn(async move {
    reader
        .fetch_cdc(keyspace, table_name, |row| consumer.consume_cdc(row))
        .await
})
```

Reply:
The `fetch_cdc` will have to be modified so that the lambda/closure is able to return `anyhow::Result<()>`.